src/main/java/de/tum/in/www1/exerciseapp/domain/DragAndDropQuestion.java

```diff
@@ -68,20 +68,84 @@ public DragAndDropQuestion dropLocations(List<DropLocation> dropLocations) {
         return this;
     }
 
+    /**
+     * 1. add the dropLocation to the List of the other dropLocation
+     * 2. add backward relation in the dropLocation-object
+     * 3. add the new dropLocation to the DragAndDropQuestionStatistic
+     *
+     * @param dropLocation the dropLocation object which will be added
+     * @return this DragAndDropQuestion-object
+     */
     public DragAndDropQuestion addDropLocations(DropLocation dropLocation) {
         this.dropLocations.add(dropLocation);
         dropLocation.setQuestion(this);
+        //if an answerOption was added then add the associated AnswerCounter implicitly
+        ((DragAndDropQuestionStatistic)getQuestionStatistic()).addDropLocation(dropLocation);
         return this;
     }
 
```

Comment:
Please reduce the number of casts in the following methods. Test if the cast is possible once, then execute the cast, then do not cast any more the same object (except it is absolutely necessary)